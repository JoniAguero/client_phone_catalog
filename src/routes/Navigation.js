import React, { useEffect } from "react"
import { BrowserRouter as Router, Route, Switch } from "react-router-dom"
import routes from "./Routes"
import { map } from "lodash"
import { useDispatch } from "react-redux"
import { isLogged } from "../redux/actions/authActions"

export default function Navigation() {
  
  const dispatch = useDispatch()

  useEffect(() => {
    dispatch(isLogged())
  }, [dispatch])

  return (
    <Router>
      <Switch>
        {map(routes, (route, index) => (
          <Route
            key={index}
            path={route.path}
            exact={route.exact}
            render={(props) => (
              <route.layout>
                <route.component {...props} />
              </route.layout>
            )}
          />
        ))}{" "}
      </Switch>
    </Router>
  )
}
