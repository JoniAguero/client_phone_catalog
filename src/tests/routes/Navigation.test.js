import React from 'react'
import { mount } from 'enzyme'
import { Provider } from 'react-redux'

import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import '@testing-library/jest-dom'
import Navigation from '../../routes/Navigation'

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

const initialState = {
  phones: {
    list: [],
    phoneSelected: null
  },
  errors: {
    error: {}
  }
}

describe('Test <Navigation /> ', () => {
  test('Match snapshot', () => {
    
    const store = mockStore(initialState)

    const wrapper = mount(
      <Provider store={store}>
        <Navigation />
      </Provider>
    )

    expect( wrapper ).toMatchSnapshot();
  })
})
