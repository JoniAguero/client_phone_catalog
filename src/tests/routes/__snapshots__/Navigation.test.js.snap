// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Navigation />  Match snapshot 1`] = `
Array [
  <div
    className="container-header"
  >
    <div
      className="link"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <h2
          className="ui header"
        >
          <i
            aria-hidden="true"
            className="mobile alternate icon"
            onClick={[Function]}
          />
          <div
            className="content"
          >
            Phone Catalog
            <div
              className="sub header"
            >
              The latest phones on the market are here
            </div>
          </div>
        </h2>
      </a>
    </div>
  </div>,
  <div
    className="ui container layout-basic"
  >
    <div
      className="container-home"
    >
      <div
        className="container-loading"
      >
        <div
          key="0"
        >
          <div
            className="ui doubling stackable three cards"
          >
            <div
              className="ui card"
              onClick={[Function]}
            >
              <div
                className="ui placeholder"
              >
                <div
                  className="square image"
                />
              </div>
              <div
                className="content"
              >
                <div
                  className="ui placeholder"
                >
                  <div
                    className="header"
                  >
                    <div
                      className="line very short"
                    />
                    <div
                      className="line medium"
                    />
                  </div>
                  <div
                    className="paragraph"
                  >
                    <div
                      className="line short"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              className="ui card"
              onClick={[Function]}
            >
              <div
                className="ui placeholder"
              >
                <div
                  className="square image"
                />
              </div>
              <div
                className="content"
              >
                <div
                  className="ui placeholder"
                >
                  <div
                    className="header"
                  >
                    <div
                      className="line very short"
                    />
                    <div
                      className="line medium"
                    />
                  </div>
                  <div
                    className="paragraph"
                  >
                    <div
                      className="line short"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              className="ui card"
              onClick={[Function]}
            >
              <div
                className="ui placeholder"
              >
                <div
                  className="square image"
                />
              </div>
              <div
                className="content"
              >
                <div
                  className="ui placeholder"
                >
                  <div
                    className="header"
                  >
                    <div
                      className="line very short"
                    />
                    <div
                      className="line medium"
                    />
                  </div>
                  <div
                    className="paragraph"
                  >
                    <div
                      className="line short"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          key="1"
        >
          <div
            className="ui doubling stackable three cards"
          >
            <div
              className="ui card"
              onClick={[Function]}
            >
              <div
                className="ui placeholder"
              >
                <div
                  className="square image"
                />
              </div>
              <div
                className="content"
              >
                <div
                  className="ui placeholder"
                >
                  <div
                    className="header"
                  >
                    <div
                      className="line very short"
                    />
                    <div
                      className="line medium"
                    />
                  </div>
                  <div
                    className="paragraph"
                  >
                    <div
                      className="line short"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              className="ui card"
              onClick={[Function]}
            >
              <div
                className="ui placeholder"
              >
                <div
                  className="square image"
                />
              </div>
              <div
                className="content"
              >
                <div
                  className="ui placeholder"
                >
                  <div
                    className="header"
                  >
                    <div
                      className="line very short"
                    />
                    <div
                      className="line medium"
                    />
                  </div>
                  <div
                    className="paragraph"
                  >
                    <div
                      className="line short"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              className="ui card"
              onClick={[Function]}
            >
              <div
                className="ui placeholder"
              >
                <div
                  className="square image"
                />
              </div>
              <div
                className="content"
              >
                <div
                  className="ui placeholder"
                >
                  <div
                    className="header"
                  >
                    <div
                      className="line very short"
                    />
                    <div
                      className="line medium"
                    />
                  </div>
                  <div
                    className="paragraph"
                  >
                    <div
                      className="line short"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
